src/Microsoft.Extensions.Logging.Configuration/LoggerProviderConfigurationFactory.cs

```diff
@@ -0,0 +1,50 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Extensions.Logging.Configuration
+{
+    internal class LoggerProviderConfigurationFactory : ILoggerProviderConfigurationFactory
+    {
+        private readonly IEnumerable<LoggingConfiguration> _configurations;
```

Comment:
This is pretty weird. Usually configuration is composed at the provider level not by using multiple IConfiguration instances. Why do we do this?

Reply:
We allow composition of logging configurations for light-up scenarios (AI is adding another logging config file), so when you call `AddConfiguration` multiple times filtering rules get flattened in one list. I just simulated the same behavior for configuration sections.